lua/neo-tree/sources/common/commands.lua

```diff
@@ -11,34 +11,42 @@ local log = require("neo-tree.log")
 local help = require("neo-tree.sources.common.help")
 local Preview = require("neo-tree.sources.common.preview")
 
----Gets the node parent folder recursively
----@param tree table to look for nodes
----@param node table to look for folder parent
+---Gets the node parent folder
+---@param state table to look for nodes
 ---@return table table
-local function get_folder_node(state, node)
+local function get_folder_node(state)
   local tree = state.tree
-  if not node then
-    node = tree:get_node()
-  end
+  local node = tree:get_node()
+  local last_id = node:get_id()
+
+  -- arbitrary number to avoid infinite loop
+  while true do
+    local insert_as_local = state.config.insert_as
+    local insert_as_global = require("neo-tree").config.window.insert_as
+    local use_parent
+    if insert_as_local then
+      use_parent = insert_as_local == "sibling"
+    else
+      use_parent = insert_as_global == "sibling"
+    end
 
-  local insert_as_local = state.config.insert_as
-  local insert_as_global = require("neo-tree").config.window.insert_as
-  local use_parent
-  if insert_as_local then
-    use_parent = insert_as_local == "sibling"
-  else
-    use_parent = insert_as_global == "sibling"
-  end
+    local is_open_dir = node.type == "directory" and (node:is_expanded() or node.empty_expanded)
+    if use_parent and not is_open_dir then
+      return tree:get_node(node:get_parent_id())
+    end
 
-  local is_open_dir = node.type == "directory" and (node:is_expanded() or node.empty_expanded)
-  if use_parent and not is_open_dir then
-    return tree:get_node(node:get_parent_id())
-  end
+    if node.type == "directory" then
+      return node
+    end
 
-  if node.type == "directory" then
-    return node
+    local parent_id = node:get_parent_id()
```

Comment:
> Something like "if node is nil or node == prior_node".

that's pretty much exactly what I did here

Reply:
I tested this change with the issue recreation and this does address the bug. A quick run through of other usages shows this to be "working", though I am not confident enough in my knowledge of the various edge cases for Neo-tree to sign off saying this is "g2g".

Reply:
yeah, I was writing my comment at the same time you were committing the fix. I think this section is good to go, with the one caveat about the `while true` condition above.